pkg/distributor/ha_tracker.go

```diff
@@ -371,7 +372,12 @@ func (h *defaultHaTracker) loop(ctx context.Context) error {
 		return true
 	})
 
```

Comment:
[nit] Above, when we call `WatchPrefix()` I would add a comment to explain that it blocks until the context has done or the callback function returns false (which can't happen because we never return false).

Reply:
I added a comment there.